akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -150,24 +150,28 @@ object Behavior {
   }
 
   /**
-   * INTERNAL API.
+   * INTERNAL API
    */
   @InternalApi
   private[akka] object UnhandledBehavior extends Behavior[Nothing] {
     override def toString = "Unhandled"
   }
 
-  /**
-   * INTERNAL API.
-   */
+  /** INTERNAL API */
   @InternalApi
   private[akka] abstract class UntypedBehavior[T] extends Behavior[T] {
-    /**
-     * INTERNAL API
-     */
+    /** INTERNAL API */
     @InternalApi private[akka] def untypedProps: akka.actor.Props
   }
 
+  // FIXME not final solution I guess
+  /** INTERNAL API: Used to bridge Behaviors between Java DSL and Scala DSL, where one of the sides is the implementation */
+  @InternalApi
+  private[akka] abstract class DelegatingBehavior[T] extends Behavior[T] {
```

Comment:
What's the difference to `Behavior.setup`?

Reply:
This has a different purpose. It is used in javadsl:

`abstract class PersistentBehavior[Command, Event, State >: Null](val persistenceId: String) extends DelegatingBehavior[Command] {`

in order to make the javadsl class "be the" scaladsl behavior:

```
override def delegate: Behavior[Command] =
    scaladsl.PersistentBehaviors.immutable[Command, Event, State](
```

I can't easily extend the scaladsl nor should I really.

Alternatives include doing some sneaky thing with both creating some shared thing... but they would not be IS-A behaviour without using such trick

Reply:
I can try to extract the `final class PersistentBehavior[Command, Event, State](` into `akka.persistence.typed` and both DLSs would extend it. That would work... I can try it, so we don't have this special case

Reply:
I don't think I can pull off `abstract class PersistentBehavior[Command, Event, State >: Null](val persistenceId: String) extends PersistentBehavior[Command, Event, State](` without very intensely changing things to be functions when.. since we need to pass things to the constructor, but those are just methods here...

I did manage to pull off a simplification, in which only one way for coexistence exists: https://github.com/akka/akka/pull/24454/commits/ab5a253f51f6d8b8473fd46e0ea750ea4525eefd